datadogreceiver/go.mod

```
@@ -0,0 +1,56 @@
+module github.com/McSick/datadogreceiver
+
+go 1.18
```

Collector contrib modules are created with go v1.17 - mixing go version could lead to problems as the collector binary is created as part of the builder process.

```suggestion
go 1.17
```